From e29bb5f295d055068a24b629fa1bd5f06a93431b Mon Sep 17 00:00:00 2001
From: Chris Wilson <chris@chris-wilson.co.uk>
Date: Sat, 06 Oct 2012 17:53:41 +0000
Subject: win32: Use the image surface below the fallback when unmapping an HDC

As for a native window, the surface does not have an image delegate
itself but instead installs a fallback surface during map_to_image. So
during unmap_image, we then need to unmap from the fallback surface
instead.

Signed-off-by: Chris Wilson <chris@chris-wilson.co.uk>
---
diff --git a/src/win32/cairo-win32-display-surface.c b/src/win32/cairo-win32-display-surface.c
index 2c5374c..b0c2f90 100644
--- a/src/win32/cairo-win32-display-surface.c
+++ b/src/win32/cairo-win32-display-surface.c
@@ -497,6 +497,7 @@ _cairo_win32_display_surface_unmap_image (void                    *abstract_surf
 		__FUNCTION__, r.x, r.y, r.width, r.height));
 	surface->fallback->damage =
 	    _cairo_damage_add_rectangle (surface->fallback->damage, &r);
+	surface = to_win32_display_surface (surface->fallback);
     }
 
     return _cairo_surface_unmap_image (surface->image, image);
--
cgit v0.9.0.2-2-gbebe
